### PR TITLE
DM-13388: Enable visit-level sky subtraction for HSC by default

### PR DESCRIPTION
**Pull request by PaulPrice**

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -238,6 +238,28 @@ preSfm = command("sfm", [skymap, transmissionCurvesTarget],
 env.Depends(preSfm, refcat)
 sfm = {(data.visit, data.ccd): data.sfm(env) for data in sum(allData.values(), [])}
 
+# Sky correction
+def processSkyCorr(dataList):
+    """Generate sky corrections"""
+    visitDataLists = defaultdict(list)
+    for ff in allData:
+        for data in allData[ff]:
+            visitDataLists[data.visit].append(data)
```
Comment (TallJimbo) [inline SConstruct]: Interesting idea to use `defaultdict`; for this pattern I always use a regular dict with `d.setdefault(key, []).append(...)`, but I think yours might be clearer.

File: SConstruct [hunk 2]
```diff
@@ -238,6 +238,28 @@ preSfm = command("sfm", [skymap, transmissionCurvesTarget],
 env.Depends(preSfm, refcat)
 sfm = {(data.visit, data.ccd): data.sfm(env) for data in sum(allData.values(), [])}
 
+# Sky correction
+def processSkyCorr(dataList):
+    """Generate sky corrections"""
+    visitDataLists = defaultdict(list)
+    for ff in allData:
+        for data in allData[ff]:
```
Comment (TallJimbo) [inline SConstruct]: Longer variable name would help here.

File: SConstruct [hunk 3]
```diff
@@ -238,6 +238,28 @@ preSfm = command("sfm", [skymap, transmissionCurvesTarget],
...
+            visitDataLists[data.visit].append(data)
+    preSkyCorr = command("skyCorr", skymap,
+                         getExecutable("pipe_drivers", "skyCorrection.py") + " " + PROC + " " + STDARGS +
+                         " --batch-type=smp --cores=1")
```
Comment (TallJimbo) [inline SConstruct]: Why `smp` rather than `none`?

File: SConstruct [hunk 4]
```diff
@@ -238,6 +238,28 @@ preSfm = command("sfm", [skymap, transmissionCurvesTarget],
...
+    nameList = ("skyCorr-%d" % (vv,) for vv in visitDataLists)
+    depList = ([sfm[(data.visit, data.ccd)] for data in visitDataLists[vv]] for vv in visitDataLists)
+    cmdList = (getExecutable("pipe_drivers", "skyCorrection.py") + " " + PROC + " " + STDARGS +
+               " --batch-type=smp --cores=1 --id visit=%d" % (vv,) for vv in visitDataLists)
+    validateList = ([validate(SkyCorrValidation, DATADIR, data.dataId) for data in visitDataLists[vv]] for
+                    vv in visitDataLists)
+    return {vv: command(name, [preSkyCorr] + dep, [cmd] + val) for vv, name, dep, cmd, val in
+            zip(visitDataLists, nameList, depList, cmdList, validateList)}
```
Comment (TallJimbo) [inline SConstruct]: Took me a while to parse this line; using kwargs in the call to `command` might help a lot.

File: SConstruct (hunk 3 above)
Comment (PaulPrice) [reply SConstruct]: With `--cores=1` there's not a lot of difference, but `none` might produce more useful error messages so I'll go with that.

File: SConstruct (hunk 4 above)
Comment (PaulPrice) [reply SConstruct]: You should have seen it before I refactored it to `zip` over the various generators!

**Pull request merged**